examples/wifi-echo/server/esp32/main/RendezvousMessageHandler.cpp

```diff
@@ -0,0 +1,55 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "RendezvousMessageHandler.h"
+#include <support/CodeUtils.h>
+#include <support/logging/CHIPLogging.h>
+#include <system/SystemPacketBuffer.h>
+
+using namespace ::chip;
+
+extern CHIP_ERROR SetWiFiStationProvisioning(char * ssid, char * key);
+
+CHIP_ERROR RendezvousMessageHandler::HandleMessageReceived(System::PacketBuffer * buffer)
+{
+    CHIP_ERROR err         = CHIP_NO_ERROR;
+    const size_t bufferLen = buffer->DataLength();
+    char * key             = nullptr;
+    char * ssid            = nullptr;
+    char msg[bufferLen];
```

Comment:
This makes the stack utilisation dependent on the received DataLength. Is the data length known to be capped to a threshold? We could be more deterministic by having the high threshold value in here?

Reply:
Good catch; I'm surprised we don't use -Wvla.

Reply:
Could you please file a followup issue to use `-Wvla`?

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/2427 proposing `-Wvla`.

(For this PR, https://github.com/project-chip/connectedhomeip/pull/2418/commits/d9dc684ccba267ceca23f532a079fc77a8d170ce removes the VLA.)